kernel/dw1000.c

```diff
@@ -2946,19 +3000,32 @@ static int dw1000_load_eui64(struct dw1000 *dw)
  */
 static int dw1000_load_delays(struct dw1000 *dw)
 {
+	uint32_t array[2];
 	int delays;
 	int rc;
 
 	/* Read delays from OTP */
 	if ((rc = regmap_read(dw->otp, DW1000_OTP_DELAYS, &delays)) != 0)
 		return rc;
-	dw->antd[DW1000_PRF_16M] = DW1000_OTP_DELAYS_16M(delays);
-	dw->antd[DW1000_PRF_64M] = DW1000_OTP_DELAYS_64M(delays);
+	if (delays != ~0) {
+		dw->antd[DW1000_PRF_16M] = DW1000_OTP_DELAYS_16M(delays);
+		dw->antd[DW1000_PRF_64M] = DW1000_OTP_DELAYS_64M(delays);
+	}
 
 	/* Read delays from devicetree */
-	of_property_read_u16_array(dw->dev->of_node, "decawave,antd",
-				   dw->antd, ARRAY_SIZE(dw->antd));
+	if (of_property_read_u32_array(dw->dev->of_node, "decawave,antd",
+				       array, ARRAY_SIZE(array)) == 0) {
+		if ((array[0] & ~DW1000_ANTD_MASK) || (array[1] & ~DW1000_ANTD_MASK)) {
+			dev_err(dw->dev, "invalid decawave,antd value <%d,%d>\n",
+				array[0], array[1]);
+			return 0;
+		}
+		dw->antd[DW1000_PRF_16M] = array[0];
+		dw->antd[DW1000_PRF_64M] = array[1];
+	}
```

Comment:
This commit should include the EEPROM generation update to match the new layout.